frame/evm/precompile/contracts/bridge/ethereum/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2021 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// --- core ---
+use core::marker::PhantomData;
+// --- crates.io ---
+use codec::Encode;
+use evm::{executor::PrecompileOutput, Context, ExitError, ExitSucceed};
+// --- darwinia-network ---
+use darwinia_evm_precompile_utils::{selector, DvmInputParser};
+use dp_evm::Precompile;
+
+#[selector]
+enum Action {
+	BurnAndRemoteUnlock = "burn_and_remote_unlock(uint32,address,address,address,bytes,uint256)",
+	TokenRegisterResponse = "token_register_response(address,address,address)",
+}
+
+/// The contract address: 0000000000000000000000000000000000000017
```

Comment:
Not in the scope of this PR, but we still need to think about the management of precompile address allocation and compatibility design.

These actions looks short life, there could be more similar actions from business level in future, and also this could be removed too in future.